controllers/upgrade.go

```diff
@@ -113,6 +121,44 @@ func (r *RollingUpgradeReconciler) ReplaceNodeBatch(rollingUpgrade *v1alpha1.Rol
 	return true, nil
 }
 
+func (r *RollingUpgradeReconciler) SetInstanceTag(rollingUpgrade *v1alpha1.RollingUpgrade, instance *autoscaling.Instance, tagKey string, tagVal string) error {
+	return awsprovider.TagEC2instance(aws.StringValue(instance.InstanceId), tagKey, tagVal, r.Auth.Ec2Client)
+}
+
+func (r *RollingUpgradeReconciler) CanTerminate(instance *autoscaling.Instance) bool {
+	instances, err := r.Auth.DescribeTaggedInstanceIDs(instanceStateTagKey, completedTagValue)
+	if err != nil {
+		r.Error(err, "unable to get tagged instances")
+		//Exit.
+	}
+
+	instanceID := aws.StringValue(instance.InstanceId)
+	if common.ContainsEqualFold(instances, instanceID) {
+		return false
+	}
+
+	if common.ContainsEqualFold(v1alpha1.ASGNonRunningInstanceStates, aws.StringValue(instance.LifecycleState)) {
+		return false
+	}
+	return true
+}
+
+func (r *RollingUpgradeReconciler) TerminateNodeInstances(rollingUpgrade *v1alpha1.RollingUpgrade, instance *autoscaling.Instance) bool {
+	instanceID := aws.StringValue(instance.InstanceId)
+	input := &autoscaling.TerminateInstanceInAutoScalingGroupInput{
+		InstanceId:                     aws.String(instanceID),
+		ShouldDecrementDesiredCapacity: aws.Bool(false),
+	}
+
+	r.Info("Terminating node-instance", "instance", instanceID)
+	_, err := r.Auth.AsgClient.TerminateInstanceInAutoScalingGroup(input)
+	if err != nil {
+		r.Error(err, "Failed to terminate instance", "name", rollingUpgrade.NamespacedName(), "instance", instanceID)
```

Comment:
can you also make sure to return the error instead of logging it here?
Let the caller decide what to do